### PR TITLE
Moved offline training snippet (from Player::onCreatureAppear) to Lua

### PR DESCRIPTION
**Pull request by ninjalulz**
Moved offline training snippet (from Player::onCreatureAppear) to Lua (Makes it easier to customizable the code).
Moved doPlayerSetOfflineTrainingSkill to compat.lua.
New functions:
* Player.addOfflineTrainingTime(addTime)
* Player.getOfflineTrainingTime(self)
* Player.removeOfflineTrainingTime(removeTime)
* Player.addOfflineTrainingTries(skillType, tries)
* Player.getOfflineTrainingSkill(self)
* Player.setOfflineTrainingSkill(skillId)

Edit:
* Fixed an issue with the absence text (there was an extra space infront of 1 hour/minute), and changed to string.format.

### CODE REVIEW

**Comment by WibbenZ** [comment]
Why not use events insted of creaturescripts?

**Comment by ninjalulz** [comment]
@WibbenZ I didn't have that in mind, but it can easily be arranged if it's preferable.

**Comment by WibbenZ** [comment]
Well im not 100% but didn't Mark or Dalkon say events are only executed when they need to, while creaturescripts are always executed or something like that?

**Comment by ranisalt** [comment]
Great addition!

**Comment by Luckey1729** [comment]
:+1: :+1: 

Thanks for the addition!

**Comment by Kamenuvol** [comment]
@WibbenZ
I think it's the opposite, events are always executed and creaturescript only when is needed.

**Comment by WibbenZ** [comment]
Now that I think that sounds wrong, events got added in the 1.x series while creaturescripts have been used for how long?
Not saying that creaturescripts are bad but why add a new system that isen't better then the old one?

Maybe Mark or Dalkon(who created it) can give us an input :p

**Comment by Nu77** [comment]
I think events was added for move core code to scripting code, anyway i think creatureevents is a deprecated way which didn't should be used.

**Comment by WibbenZ** [comment]
@ninjalulz 
https://github.com/ninjalulz/forgottenserver/blob/OfflineTraining/data/actions/scripts/other/skilltrainer.lua

File: src/luascript.cpp [hunk 1]
```diff
@@ -8306,6 +8297,89 @@ int LuaScriptInterface::luaPlayerAddSkillTries(lua_State* L)
...
+}
+
+int LuaScriptInterface::luaPlayerAddOfflineTrainingTries(lua_State* L)
+{
+	// player:addOfflineTrainingTries(skillType, tries)
+	Player* player = getUserdata<Player>(L, 1);
+	if (player) {
+		skills_t skillType = getNumber<skills_t>(L, 2);
+		uint64_t tries = getNumber<uint64_t>(L, 3);
+		player->addOfflineTrainingTries(skillType, tries);
```
Comment (marksamman) [inline src/luascript.cpp]: This branch must push something to the Lua stack or return 0. It causes undefined behavior to not return the same number as objects you're pushing to the Lua stack (usually causes LuaJIT to crash).

Comment (marksamman) [inline src/luascript.cpp]: Looks like it should be:
```cpp
pushBoolean(L, player->addOfflineTrainingTries(skillType, tries));
```

Comment (ninjalulz) [inline src/luascript.cpp]: Thanks, I totally missed that! Fixed.

**Comment by marksamman** [comment]
Thanks!

**Pull request merged**